### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by vitaly-t** [Klathmon/PCS-API#1]
According to the search statistics on GitHub, you are the only developer who chose to use option `pgFormatting` for the pg-promise library.

May I ask, why you thought it would be a better option for your project? I'm just curious, to see whether or not I should consider this option as obsolete in the future.

**Comment by Klathmon** [comment]
To be honest it's because I don't need any of the additional formatting options provided by pg-promise, and when in doubt I try not to use features I don't need.

Plus I (maybe wrongly, i'm not sure) assumed that the node-postgres would handle it faster, if there were any bugs that the larger library would have a better chance at getting them fixed for me, and it was unclear if i could use prepared statements (which was a must for me).

If you added some clarification about any possible performance issues (or lack of) to your readme and clarified if prepared statements are still usable, i would have no issues if the option went away.

But if prepared statements can't be used and this option goes away i wouldn't be very happy...

**Comment by vitaly-t** [comment]
Thank you for your feedback!

I continuously improve the documentation, see [wiki](https://github.com/vitaly-t/pg-promise/wiki).

About the prepared statements, they are not effected by the formatting option, as they stand away from all that, i.e. formatting for the prepared statements is always handled by node-postgres.

In terms of the speed, in this case is also irrelevant, because by the very nature, prepared statements are formatted on the database server side, that's how it prevents SQL injection.

**Comment by Klathmon** [comment]
I meant more the parsing speed of your library (walking through the string to look for named params and stuff), although a quick test show it doesn't affect anything. (I just wanted to make sure that it wouldn't cause issues if i went and inserted 1000 lines in one big query or something)

So i wouldn't have any issues if it was dropped, as it looks like i probably didn't fully understand what it did anyway.

**Comment by vitaly-t** [comment]
Parsing speed is quite irrelevant. When it comes to a database library like this, no matter what code you write, at least 99% of time will be spent inside the IO, i.e. waiting for response from the port once the query has been submitted to the server.

On another side though, I saw some ORM-s manage to add so much overhead, their speed is dropped. One of the many reasons for staying away from ORM-s :)

I shared some thought on that here: [Node.js and PostgreSQL](http://blog.vitalytomilov.com/pg/postgresql-and-nodejs/)
